Src/WitsmlExplorer.Frontend/components/Sidebar/MudLogItem.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import NavigationContext from "../../contexts/navigationContext";
+import NavigationType from "../../contexts/navigationType";
+import GeologyInterval from "../../models/geologyInterval";
+import MudLog from "../../models/mudLog";
+import { calculateObjectNodeId } from "../../models/objectOnWellbore";
+import { ObjectType } from "../../models/objectType";
+import Well from "../../models/well";
+import Wellbore, { getGeologyIntervalLength } from "../../models/wellbore";
+import GeologyIntervalItem from "./GeologyIntervalItem";
+import TreeItem from "./TreeItem";
+import { WellboreItemContext } from "./WellboreItem";
+import { ComponentType } from "../../models/componentType";
+import ComponentService from "../../services/componentService";
+import ModificationType from "../../contexts/modificationType";
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+interface ObjectOnWellboreItemProps {
+  mudlogData: MudLog;
+  objectType: ObjectType;
+}
+
+const MudLogItem = (props: ObjectOnWellboreItemProps): React.ReactElement => {
+  const { mudlogData, objectType } = props;
+
+  const { wellbore, well } = useContext(WellboreItemContext);
+  const { dispatchNavigation, navigationState } = useContext(NavigationContext);
+  const { selectedObject, selectedObjectGroup } = navigationState;
+
+  const isSelected = useCallback(
+    (mudlog: MudLog) => {
+      return selectedObject &&
+        selectedObjectGroup === ObjectType.MudLog &&
+        selectedObject.mudloguid === mudlog.uid &&
+        selectedObject.wellboreUid === mudlog.wellboreUid &&
+        selectedObject.wellUid === mudlog.wellUid
+        ? true
+        : undefined;
+    },
+    [selectedObject, selectedObjectGroup]
+  );
+
+  return (
+    <>
+      <TreeItem
+        nodeId={calculateObjectNodeId(mudlogData, ObjectType.MudLog)}
+        labelText={mudlogData.name}
+        onLabelClick={() => {
+          dispatchNavigation({
+            type: NavigationType.SelectObject,
+            payload: {
+              object: mudlogData,
+              wellbore: wellbore,
+              well: well,
+              objectType
+            }
+          });
+        }}
+      >
+        {listLogItemsByType(mudlogData, well, wellbore, isSelected)}
+      </TreeItem>
+    </>
+  );
+};
+
+const listLogItemsByType = (
+  mudlogs: MudLog,
+  well: Well,
+  wellbore: Wellbore,
+  isSelected: (mudlogs: MudLog) => boolean
+) => {
+  const { dispatchNavigation } = useContext(NavigationContext);
+  const [geologyIntervals, setGeologyIntervals] = useState<GeologyInterval[]>(
+    []
+  );
+
+  const getGeologyInterval = () => {
+    const abortController = new AbortController();
+    const getGeologyIntervals = async () => {
+      const geologyresult = await ComponentService.getComponents(
```

Comment:
I'm not sure if this is the correct location causing the bug, but when there is a mud log that is not expandable we get into a situation where the fetching goes into a infinity loop:
![image](https://github.com/equinor/witsml-explorer/assets/48485965/c6f36e8b-c9e3-4dfa-be8e-76572ce46ec7)
![image](https://github.com/equinor/witsml-explorer/assets/48485965/d0826142-8215-4f41-b502-8e3d9a071eeb)